src/com/google/javascript/jscomp/CommandLineRunner.java

```diff
@@ -867,12 +867,21 @@
         name = "--instrument_code",
         usage =
             "Enable code instrumentation to perform code coverage analysis. Options are:\n"
-                + " 1. NONE (deault)\n"
+                + " 1. NONE (default)\n"
                 + " 2. LINE - Instrument code by line.\n"
                 + " 3. BRANCH - Instrument code by branch.\n"
                 + " 4. PRODUCTION - Function Instrumentation on compiled JS code.\n")
     private String instrumentCode = "NONE";
 
+    @Option(
+        name = "--production_instrumentation_array",
+        usage =
+            "Name of the global array used by production instrumentation. The array name "
+                + "should be declared as an extern so it is not renamed by the compiler. A function"
```

Comment:
The name wouldbe an extern so that it wont be renamed by the compiler. The engineer would then be able to make their function based on this name and know that it will work without undefined errors. If it did cause errors during the checks phase, then we could also just move the declaration of the array before any instrumentation is done. Although I wasn't aware that this could be an issue.

As for flag name, I do agree on the length of the flag and I couldn't come up with a better one. What about `prod_instr_array_name` I don't personally like the short form use but it is shorter.

Reply:
SGTM.